libs/crypto/crypto-impl/src/main/kotlin/net/corda/crypto/impl/retrying/CryptoRetryingExecutor.kt

```
@@ -29,50 +20,15 @@ open class CryptoRetryingExecutor(
      */
     @Suppress("NestedBlockDepth")
     fun <R> executeWithRetry(block: () -> R): R {
-        var attempt = 1
-        var op = ""
-        while (true) {
-            try {
-                if (attempt > 1) {
-                    logger.info("Retrying operation (op={},attempt={})", op, attempt)
-                }
-                val result = execute(block)
-                if (attempt > 1) {
-                    logger.info("Retrying was successful (op={},attempt={})", op, attempt)
-                }
-                return result
-            } catch (e: Throwable) {
-                if (!e.isRecoverable()) {
-                    // the exception is not recoverable, no point in retrying
-                    logCompleteFailure(attempt, op)
-                    // throws the original exception
-                    throw e
-                }
-                val backoff = strategy.getBackoff(attempt)
-                if (backoff < 0 || attempt > CRYPTO_MAX_ATTEMPT_GUARD) {
```

We've lost this behaviour, which is we previously ignore a  config request to have more than 10 retries and stick at 10. That could matter and would be worth release notes.

---

Agreed, but should we even bother to keep the old behaviour in place?. What's the point of allowing users to set something higher than `10` if we are going to silently ignore it underneath?. Instead of that, I've just opened a `PR` to explicitly set the limits within the configuration directly: https://github.com/corda/corda-api/pull/1430.

---

Agreed